### ISSUE TITLE
MongoDB is not mandatory anymore?

### ISSUE THREAD

**Issue by Mte90** [perftools/xhgui#364]
On VVV now we are using Sqlite but the composer.json of this project and composer.lock require the mongo library.

If mongo is not installed the library block the execution of XHGUI with a warning and blank page, so I had to remove it from the composer.json file to work.

**Comment by glensc** [comment]
PDO support is not complete, so MongoDB is still the recommended and default datastore:
- https://github.com/perftools/xhgui/issues/320

Please include the exact error and details of your second statement in the issue body.

**Comment by Mte90** [comment]
![Screenshot_20201028_174055](https://user-images.githubusercontent.com/403283/97467780-d364c800-1944-11eb-9bec-3c1a1c8544ab.png)

This is what happens and removing the package fix it.

**Comment by glensc** [comment]
That message is emitted by composer. And removing what?

**Comment by Mte90** [comment]
Ops sorry I forgot the most important part....
Removing `alcaeus/mongo-php-adapter` from the composper.json and a composer update that warning is gone and I can use XHgui.

**Comment by glensc** [comment]
You can disable platform check or use composer 1.x:
- https://getcomposer.org/doc/06-config.md#platform-check

**Comment by Mte90** [comment]
Seems that composer v2 is enabled as true but with false is fixed.

I am trying with `composer config platform-check false`.

**Comment by tomjn** [comment]
Would it make more sense to declare the mongodb adapter package a suggested/recommended dependency? Keeping in mind we already use composer v1.

Also, changing the platform-check option appears to be a Composer v2 feature, this is the result of Composer v1:

```
vagrant@vvv:/srv/www/default/xhgui$ composer config platform-check false

                                                                             
  [InvalidArgumentException]                                                 
  Setting platform-check does not exist or is not supported by this command  
                                                                             

config [-g|--global] [-e|--editor] [-a|--auth] [--unset] [-l|--list] [-f|--file FILE] [--absolute] [--] [<setting-key>] [<setting-value>]...
```

The readme, repo description, and documentation claim that PDO is currently a supported option. It is clear that this is not the case. I suggest these be updated to reflect that PDO support is a work in progress. _We have wasted a lot of valuable time as a result of this inaccurate information._

**Comment by glensc** [comment]
Patches in form of pull requests are welcome.

**Comment by tomjn** [comment]
As is ours, I would produce a patch myself but I am unfamiliar with the conventions and structure of this repo, nor do I have the administrator access to amend all instances that need adjusting, e.g. the repo description

**Comment by glensc** [comment]
as noted in the first response in this issue, pdo support is just incomplete:
- https://github.com/perftools/xhgui/issues/364#issuecomment-718059271

"incomplete" does not mean outright wrong and pdo not working.

there have been fixes continuously to improving pdo support with releases 0.10..0.16:
- https://github.com/perftools/xhgui/releases

those fixes are based on user reports and some of the differences came up when trying to setup PHPUnit tests for pdo.

I've tried to map differences in:
- https://github.com/perftools/xhgui/issues/320

so far the "pdo" users have not complained, so I guess most of the wanted functionality is covered. the other pdo than sqlite support was fixed only in 0.15.0 really, but first pdo support was merged in 0.10.0.

**Comment by tomjn** [comment]
You have at least 2 PDO users complaining here. I don't mind if using PDO only gets us 90% of the functionality, or if we have to take additional steps, **as long as it's clearly stated**, which it is not. You yourself reviewed our PR's trying to implement PDO and did not catch this.

You can make whatever decisions you want for your own project, just that you document them accurately. I shouldn't have to read through pages of github issues on the off-chance one of them contains relevant information. It's an unreasonable expectation. ( You could even have pinned that issue so it was at the top )
